### PR TITLE
Speed up duplicate policy statement removal

### PR DESCRIPTION
**Pull request by donatello**
Duplicate detection now always runs in `O(n^2)` time (n = number of statements). Previously this could take `O(n^3)` time (when the number of duplicates is high).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/fatih/color v1.14.1
 	github.com/fatih/structs v1.1.0
 	github.com/go-ldap/ldap/v3 v3.4.4
-	github.com/lestrrat-go/jwx v1.2.25
+	github.com/lestrrat-go/jwx v1.2.26
```
Comment (harshavardhana) [inline go.mod:10]: @donatello send a separate PR for dependency update?

Comment (donatello) [reply go.mod:10]: ok sure

Comment (donatello) [reply go.mod:10]: This is done now, but CI will fail without the security fix.

**Review by harshavardhana — APPROVED** [review]

**Review by poornas — APPROVED** [review]
